Jenkinsfile

```diff
@@ -461,8 +513,24 @@ def dotnet(Closure body){
 }
 
 def dotnetWindows(Closure body){
-  def dockerTagName = 'docker.elastic.co/observability-ci/apm-agent-dotnet-windows:latest'
-  bat label: 'Docker Build', script: "docker build --tag ${dockerTagName}  -m 2GB .ci\\docker\\buildtools-windows"
+  def dockerUri = 'docker.elastic.co/';
+  def dockerTagName = "${dockerUri}observability-ci/apm-agent-dotnet-windows:latest"
+  dockerLogin(secret: "secret/apm-team/ci/docker-registry/prod",
+  registry: "docker.elastic.co")
+  // Remove this line once push/pull is enabled by Infra
+  dockerTagName = 'docker.io/elwpenn/windows-vstudio-msbuild:latest'
+  try {
```

Comment:
NIT:
This could be moved to the sh step with the returnStatus and then some if conditions:

```
     def status = bat(label: 'Docker Pull', script: "docker pull ${dockerTagName}", returnStatus: true)
     if (status > 0) {
        status = bat(label: 'Docker Build', script: "docker build --tag ${dockerTagName} -m 2GB .ci\\docker\\buildtools-windows", returnStatus: true)
     }
     if (status > 0) {
        status = bat(label: 'Docker Push', script: "docker push ${dockerPushTag}", returnStatus: true)
     }
     if (status > 0) {
        log(level: "WARNING", text: 'Push didn't work, we can still build despite this.')
     }
```

What are your thoughts?

Reply:
IIRC it takes more than 30min to build the image, build the image I think is not an option.

Reply:
The image should ever only build if the image isn't available to be pulled.  Need to figure out if we ever actually need this image rebuilt and setup a condition for this to happen outside of the build pipeline.  For the moment this will work to initially seed the image and allow us to pull the image from docker registry.